encoder/src/main/java/openlr/encoder/postprocessor/OfframpProcessor.java

```diff
@@ -0,0 +1,97 @@
+package openlr.encoder.postprocessor;
+
+import openlr.OpenLRProcessingException;
+import openlr.encoder.properties.OpenLREncoderProperties;
+import openlr.map.GeoCoordinates;
+import openlr.map.Line;
+import openlr.map.Node;
+import openlr.map.utils.GeometryUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class OfframpProcessor extends AbstractLrpProcessor {
+
+    private static double BEARING_RANGE_OF_SIBLING_LINES = 45;
+
+    public static class RouteNodeInfo {
+        public Node node;
+        public double lengthViaNode;
+        public RouteNodeInfo(Node node, double lengthViaSNode) {
+            this.node = node;
+            this.lengthViaNode = lengthViaSNode;
+        }
+    }
+
+    private OfframpProcessor(OpenLREncoderProperties properties) {
+        this.properties = properties;
+    }
+
+    public static OfframpProcessor with(OpenLREncoderProperties properties) {
+        return new OfframpProcessor(properties);
+    }
+
+
+    private RouteNodeInfo getRelevantRouteNode(GeoCoordinates source, Line line, GeoCoordinates destination) {
+        GeoCoordinates startNodePosition = line.getStartNode().getGeoCoordinates();
+        double sourceToNode = GeometryUtils.distance(source, startNodePosition);
+        double nodeToDestination = GeometryUtils.distance(startNodePosition, destination);
+        double airLineDistanceViaNode = sourceToNode + nodeToDestination;
+        return new RouteNodeInfo(line.getStartNode(), airLineDistanceViaNode);
+    }
+
+    private boolean isOffRampPossible(double bearingOfLrpLine, List<Line> siblingsOfLrpLine) {
+        for (Line line : siblingsOfLrpLine) {
+            double bearingOfSiblingLine = GeometryUtils.calculateLineBearing(line, GeometryUtils.BearingDirection.IN_DIRECTION, this.properties.getBearingDistance(), 0);
+            double difference = GeometryUtils.bearingDifference(bearingOfLrpLine, bearingOfSiblingLine);
+            if (difference <= BEARING_RANGE_OF_SIBLING_LINES) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @return true if postprocessor is not configured in properties file;
+     */
+    protected final boolean isNotActive() {
+        return !properties.insertLrpToAvoidOffRamp();
+    }
+
+    public List<Integer> determineNewIntermediatePoints(List<Line> route) throws OpenLRProcessingException {
+
```

Comment:
You can get rid of this new line here.

Reply:
f6c41f62659ef9375da2b7438f0985e9587cb108